repos/c2sm/packages/icon/package.py

```diff
@@ -708,9 +708,15 @@ def build(self, spec, prefix):
             make.jobs = make_jobs
         make(*self.build_targets)
 
+    def check(self):
+        # By default "check" calls make with targets "check" and "test".
+        # This testing is beyond the scope of BuildBot test at CSCS.
+        # Therefore override this function, saves a lot of time too.
```

Comment:
@jonasjucker is this description accurate from https://github.com/C2SM/spack-c2sm/issues/760 my understanding, is the same test were called twice. Should the comment not be:
```
        # By default "check" calls make with targets "check" and "test".
        # By override this function, the test is only run once when used with Builbot at CSCS
        # this is a significant gain of time in the CI
```
or are there effectively test which are not done anymore ?

Reply:
Where are "make check" and "make test" executed now?

Reply:
With BB at DKRZ make check and make test are not executed at CSCS atm. So I don't think it make sense to to more extensive testing in spack-c2sm.
Before this PR the tolerance tests for ICON were executed twice, but make check and make check were not executed at all.
So this PR removed this double-execution of these test.

Note that only c2sm-hosted ICON is affected by this change, icon-nwp is not affected because it is missing the test.py script.

Does this clarify you questions?

Reply:
No, not yet.
I thought before this PR, `spack install --test=root icon @nwp-master` executed `make check` and `make test` through spack's `autotools.check`. But this PR suppresses it. So what triggers `make check` and `make test` now?

Reply:
Also before this PR make check was not executed because the check function was overridden by the check function in Icon recipe.

This was like that since I first introduced testing for Icon

Reply:
If you wish you can remove the dummy check function and stick to the default spack provides. But as I said BB does not run these test at CSCS. Better we activate it there first...